src/flows/pipes/Notification/view.tsx

```diff
@@ -0,0 +1,415 @@
+// Libraries
+import React, {FC, useContext, useCallback, useMemo, useEffect} from 'react'
+import {parse, format_from_js_file} from '@influxdata/flux'
+import {
+  ComponentStatus,
+  Form,
+  FlexBox,
+  Input,
+  InputType,
+  Icon,
+  IconFont,
+  ComponentSize,
+  Tabs,
+  Orientation,
+  TextArea,
+  AlignItems,
+} from '@influxdata/clockface'
+import {RemoteDataState} from 'src/types'
+
+import {PipeContext} from 'src/flows/context/pipe'
+import {PopupContext} from 'src/flows/context/popup'
+import {FlowQueryContext} from 'src/flows/context/flow.query'
+import {SidebarContext} from 'src/flows/context/sidebar'
+import {remove} from 'src/flows/context/query'
+
+import Threshold, {
+  THRESHOLD_TYPES,
+} from 'src/flows/pipes/Notification/Threshold'
+import {DEFAULT_ENDPOINTS} from 'src/flows/pipes/Notification/Endpoints'
+import ExportTaskButton from 'src/flows/pipes/Schedule/ExportTaskButton'
+import ExportTaskOverlay from 'src/flows/pipes/Schedule/ExportTaskOverlay'
+
+// Types
+import {PipeProp} from 'src/types/flows'
+
+// Utils
+import {event} from 'src/cloud/utils/reporting'
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
+
+const Notification: FC<PipeProp> = ({Context}) => {
+  const {id, data, queryText, update, range, results, loading} = useContext(
+    PipeContext
+  )
+  const {register} = useContext(SidebarContext)
+  const {launch} = useContext(PopupContext)
+  const {simplify} = useContext(FlowQueryContext)
+
+  let intervalError = ''
+  let offsetError = ''
+
+  if (!data.interval) {
+    intervalError = 'Required'
+  } else if (
+    data.interval !==
+    data.interval.match(/(?:(\d+(y|mo|s|m|w|h){1}))/g)?.join('')
+  ) {
+    intervalError = 'Invalid Time'
+  }
+
+  if (
+    data.offset &&
+    data.offset !== data.offset.match(/(?:(\d+(y|mo|s|m|w|h){1}))/g)?.join('')
+  ) {
+    offsetError = 'Invalid Time'
+  }
+
+  const hasTaskOption = useMemo(
+    () =>
+      !!Object.keys(
+        remove(
+          parse(simplify(queryText)),
+          node =>
+            node.type === 'OptionStatement' &&
+            node.assignment.id.name === 'task'
+        ).reduce((acc, curr) => {
+          // eslint-disable-next-line no-extra-semi
+          ;(curr.assignment?.init?.properties || []).reduce((_acc, _curr) => {
+            if (_curr.key?.name && _curr.value?.location?.source) {
+              _acc[_curr.key.name] = _curr.value.location.source
+            }
+
+            return _acc
+          }, acc)
+
+          return acc
+        }, {})
+      ).length,
```

Comment:
Is this really the only way to see if there's a task option?

Reply:
the only good way that doesn't break